build/Makefile.simple

```diff
@@ -31,6 +31,9 @@ clean:
 		rmdir bin --ignore-fail-on-non-empty; \
 	fi
 
+%.s: %.S
```

Comment:
Can you tell me what this does? And presumably this doesn't affect any of the other examples.

Reply:
It overrides the default recipe for compiling `.S` files into `.s` files. The default recipe runs the preprocessor on an `.S` file to obtain the corresponding `.s`, but it completely ignores the `CFLAGS`. I want it to use the `CFLAGS` (to which I appended the relevant include paths), because that allows us to remove the relative paths from any `#include`s we might have in our `.S` files (it allows us to go from `#include ../../../../../include/common.h` to `#include common.h`)

Reply:
It shouldn't affect any of the other examples (the recipe is the mostly same as before, the only difference being that it uses the `CFLAGS` we define)